app/_how-tos/configure-oidc-with-cert-bound-tokens.md

```diff
@@ -0,0 +1,352 @@
+   ```
+1. Import the root certificates to the `.p12` file:
+{% capture "import-root" %}
+```sh
+keytool -importkeystore \
+  -deststorepass $KEYSTORE_PASS \
+  -destkeypass $KEYSTORE_PASS \
+  -destkeystore keycloak-keystore.jks \
+  -srckeystore keycloak-keystore.p12 \
+  -srcstoretype PKCS12 \
+  -srcstorepass $PKCS12_PASSWORD \
+  -alias keycloak
+```
+{% endcapture %}
+{{ import-root | indent: 3 }}
+
+1. Configure Keycloak to trust certificates signed by the CA:
+{% capture "keycloak-trust" %}
+```sh
+keytool -import -alias rootca \
+  -keystore keycloak-truststore.p12 \
+  -storetype PKCS12 \
+  -file rootCA.crt \
+  -storepass "$PKCS12_PASSWORD"
+
+keytool -list -keystore keycloak-truststore.p12 -storepass "$PKCS12_PASSWORD"
+```
+{% endcapture %}
+{{ keycloak-trust | indent: 3 }}
+Type `y` when prompted to trust the certificate. The Keycloak server presents this certificate to the client.
+
+## Configure Keycloak
+
+1. In a new terminal window, export your trust store password and hostname:
+   ```sh
+   export PKCS12_PASSWORD='YOUR-PASSWORD'
+   export HOSTNAME='YOUR-KEYCLOAK-HOSTNAME'
+   ```
+
+1. Then, start Keycloak in Docker:
+   
+   {:.danger}
+   > **Important:** In this tutorial, use your DNS hostname in place of `$HOSTNAME`.
+{% capture "keycloak-docker" %}
+```sh
+docker run \
+  -p 9443:9443 \
+  --network kong-quickstart-net \
+  -v $(pwd)/oidc/certs:/opt/keycloak/ssl \
+  -e KC_BOOTSTRAP_ADMIN_USERNAME=admin \
+  -e KC_BOOTSTRAP_ADMIN_PASSWORD=admin \
+  --name keycloak \
+  quay.io/keycloak/keycloak \
+  start \
+  --https-port=9443 \
+  --https-certificate-file=/opt/keycloak/ssl/keycloak.crt \
+  --https-certificate-key-file=/opt/keycloak/ssl/keycloak.key \
+  --https-trust-store-file=/opt/keycloak/ssl/keycloak-truststore.p12 \
+  --https-trust-store-password=$PKCS12_PASSWORD \
+  --https-client-auth=request \
+  --hostname=$HOSTNAME
+```
+{% endcapture %}
+{{ keycloak-docker | indent: 3 }}
+
+1. Open the Keycloak admin console.
+   The default URL of the console is `https://$HOSTNAME:8080/admin/master/console/`.
+1. In the sidebar, open **Clients**, then click **Create client**.
+1. Configure the client:
+{% capture "keycloak-client" %}
+<!--vale off-->
+{% table %}
+columns:
+  - title: Section
+    key: section
+  - title: Settings
+    key: settings
+rows:
+  - section: "**General settings**"
+    settings: |
+      * Client type: **OpenID Connect**
+      * Client ID: any unique name, for example `kong`
+  - section: "**Capability config**"
+    settings: |
+      * Enable **Client authentication**
+      * Enable **Authorization**
+      * Select **Standard flow**, **Direct access grants**, and **Service accounts roles** 
+{% endtable %}
+<!--vale on-->
+{% endcapture %}
+{{ keycloak-client | indent: 3 }}
+1. Click the **Credentials** tab.
+1. Select "X509 Certificate" from the **Client Authenticator** dropdown menu.
+1. Enter `CN=client-app, OU=Dev, O=ClientOrg, L=City, ST=State, C=US` in the **Subject DN** field.
```

Comment:
```suggestion
1. Enter `CN=client-app, OU=Dev, O=ClientOrg, L=City, ST=State, C=US` in the **Subject DN** field.
1. Click **Save** and agree to the change.
```
In case they try to switch to the tab before saving, they should know that there's a dialog they need to agree to first.